src/main/java/info/batey/kafka/unit/KafkaUnit.java

```diff
@@ -108,13 +109,23 @@ public void startup() {
         zookeeper = new Zookeeper(zkPort);
         zookeeper.startup();
 
-        File logDir;
+        final File logDir;
         try {
             logDir = Files.createTempDirectory("kafka").toFile();
         } catch (IOException e) {
             throw new RuntimeException("Unable to start Kafka", e);
         }
         logDir.deleteOnExit();
+        Runtime.getRuntime().addShutdownHook(new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    FileUtils.deleteDirectory(logDir);
+                } catch (IOException e) {
+                    LOGGER.warn("Problems deleting temporary directory {}.", logDir.getAbsolutePath(), e);
```

Comment:
You can't (I think) combine placeholders with a stacktrace We'll lose the exception here